File: leaf-system/src/main/java/com/leaf/domain/LeafDict.java
package com.leaf.domain;

import org.babyfish.jimmer.sql.*;

import java.util.List;

/**
 * @author jhlz
 * @version 1.0.0
 */
@Entity
public interface LeafDict extends BaseEntity, ID {

    /**
     * 字典编码
     */
    String dictCode();

    /**
     * 字典名称
     */
    String dictName();

    /**
     * 状态：0正常，1禁用
     */
    int status();

    @OneToMany(mappedBy = "dict")
    List<LeafDictItem> items();
}

File: src/main/kotlin/com/jhlz/domain/LeafDict.kt
package com.jhlz.domain

import org.babyfish.jimmer.sql.*

/**
 * @author jhlz
 * @version 1.0.0
 */
@Entity
interface LeafDict : BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    /**
     * 字典类型标识
     */
    val dictType: String

    /**
     * 字典类型名称
     */
    val dictName: String?

    /**
     * 状态：0正常，1禁用
     */
    val status: Int

    /**
     * 排序
     */
    val sortable: Int

    @OneToMany(mappedBy = "dict")
    val items: List<LeafDictItem>
}
